### PR TITLE
Feature/stk 1035 batch read and eval requests

### PR DESCRIPTION
**Pull request by garethj2**
@riccardolederj2 I've added batch processing capabilities to `haystack-nclient`. 

The first application of this is for eval (automatically batching to `evalAll`. I've added a new `read` method to the extension ops that uses eval for the batching you require.

@rracariu we need to add a new HTTP op for batching of read request to our new FIN project. Then it's easy for me to add batch processing.

I did look into combining haystack filters into one super filter and then rerunning the filters client side. However this won't always work under certain conditions (i.e. pathing to records that aren't in the client result set). Therefore it's all restricted to eval and Axon to start with. The good news is we have `BatchProcessor` which makes this easy to apply elsewhere.

### CODE REVIEW

**Review by EliteScientist — APPROVED** [review]

**Comment by EliteScientist** [comment]
LGTM

File: src/util/BatchProcessor.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+ *   processor.invoke('bar'),
+ *   processor.invoke('today')
+ * ])
+ * ```
+ */
+export class BatchProcessor<ArgType, ReturnType> {
+	/**
+	 * The internal timer id for keeping track of async activity.
+	 */
+	#timerId?: NodeJS.Timeout
```
Comment (rracariu) [inline src/util/BatchProcessor.ts]: How is this working with Deno and browsers?

**Review by rracariu — APPROVED** [review]
LGTM

File: src/util/BatchProcessor.ts (hunk 1 above)
Comment (garethj2) [reply src/util/BatchProcessor.ts]: A better way to deal with the timeout type is to use...

    ReturnType<typeof setTimeout>

https://stackoverflow.com/questions/45802988/typescript-use-correct-version-of-settimeout-node-vs-window

**Pull request merged**